### PR TITLE
Fix jax.numpy.hsplit

### PR DESCRIPTION
**Pull request by xoiga123**
[numpy docs](https://numpy.org/doc/stable/reference/generated/numpy.hsplit.html): 
> hsplit is equivalent to split with axis=1, the array is always split along the second axis except for 1-D arrays, where it is split at axis=0.

### CODE REVIEW

**Review by jakevdp — APPROVED** [review]
Thanks for catching this! This looks like a good way to fix the issue.

Before we merge, can you squash your changes into a single commit? See https://jax.readthedocs.io/en/latest/contributing.html#single-change-commits-and-pull-requests

Thanks!

**Comment by jakevdp** [comment]
It looks like there are some typing and test failures - please take a look!

Alternatively, if you'd like me to dig into it instead, please let me know.

**Comment by xoiga123** [comment]
Ah my bad, I'll fix that ASAP

**Comment by xoiga123** [comment]
@jakevdp An alternative fix is to use `_axis = 0` and pass in `_split(..., axis=_axis)`, so tell me if you don't like using `nonlocal` in the codebase

**Review by jakevdp — APPROVED** [review]
Looks good, thanks!

**Comment by xoiga123** [comment]
My bad again, missed the typing error 😓

**Comment by xoiga123** [comment]
@jakevdp sorry for not running `pre-commit` the last few times, should be fine now

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -1167,10 +1167,8 @@ def broadcast_to(array: ArrayLike, shape: Shape) -> Array:
   return _broadcast_to(array, shape)
 
 
-def _split(op: str, ary: ArrayLike, indices_or_sections: Union[int, ArrayLike],
+def _split(op: str, ary: Array, indices_or_sections: Union[int, ArrayLike],
            axis: int = 0) -> List[Array]:
-  _check_arraylike(op, ary)
-  ary = asarray(ary)
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:1173]: Since this check must be done by every wrapper that calls this helper, wouldn't it make more sense to continue to do the check just once here?

Comment (jakevdp) [reply jax/_src/numpy/lax_numpy.py:1173]: If it's a type checking issue, then you can address it by changing the variable name when you cast to array. e.g.
```python
a = asarray(ary)
```
Then `a` is of type `Array` and you can do things like `a.shape` without mypy flagging it as incorrect.

Comment (xoiga123) [reply jax/_src/numpy/lax_numpy.py:1173]: I was kinda scared of that due to creating an extra array, but that does seem cleaner

**Review by jakevdp — APPROVED** [review]
Looks good, thanks!
